fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt

```diff
@@ -251,4 +284,44 @@ sealed class AppAction : Action {
             val expanded: Boolean,
         ) : ShoppingAction()
     }
+
+    /**
+     * Actions related to the home screen.
+     */
+    sealed class ToolbarAction : AppAction() {
+        /**
+         * App should be opened to the home screen in [BrowsingMode.Normal].
+         */
+        object NewTab : ToolbarAction()
+
+        /**
+         * App should be opened to the home screen in [BrowsingMode.Private].
+         */
+        object NewPrivateTab : ToolbarAction()
+    }
+
+    /**
+     * Actions related to the home screen.
+     */
+    sealed class TabsTrayAction : AppAction() {
+        /**
+         * App should be opened to the home screen in [BrowsingMode.Normal].
+         */
+        object NewTab : TabsTrayAction()
+
+        /**
+         * App should be opened to the home screen in [BrowsingMode.Private].
+         */
+        object NewPrivateTab : TabsTrayAction()
+    }
```

Comment:
Comments might be too generic here since there's no direct reference to what is happening from Toolbar or Tabs Tray.